test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
Regarding @tomalec's example, the line `palindromModel.reviewObj = reviewState;` is incorrect now, because the server would reject such patch.

If the server allows such patches in future, can we solve this problem then? Or will it be too late and we have to make a decision now?